private-templates-service/client/src/components/Dashboard/Dashboard.tsx

```diff
@@ -26,13 +28,23 @@ interface DashboardProps {
 }
 
 class Dashboard extends React.Component<DashboardProps, DashboardStates> {
+  constructor(props: DashboardProps) {
+    super(props);
+    this.state = { error: null, isPreviewOpen: false };
+  }
+
+  toggleModal = () => {
+    this.setState({ isPreviewOpen: !this.state.isPreviewOpen });
+  }
+
   render() {
     return (
       <DashboardContainer>
         <Title>Recent</Title>
-        <Gallery></Gallery>
+        <Gallery toggleModal={this.toggleModal}></Gallery>
         <Title>Drafts</Title>
-        <Gallery></Gallery>
+        <Gallery toggleModal={this.toggleModal}></Gallery>
+        <PreviewModal show={this.state.isPreviewOpen} toggleModal={this.toggleModal} />
```

Comment:
Cant you just swap out the content rather than having a modal?
Just have the state in the dashboard of what you want to display
`{this.state.modalOpen ? <PreviewModal.... /> : <Gallery ... />`

Reply:
If you control the children and just have one rendered at a time, then the modal doesn't need to worry about hiding the stuff that was there before (and you'll run into less issues like the small screen issue below)

Reply:
Resolved in teams. The main takeaway was that we want to navigate back to the same spot on the dashboard after closing the preview page, meaning both have to be rendered simultaneously, with the preview page overlayed on top of the main dashboard page.